src/azure-cli/azure/cli/command_modules/monitor/operations/autoscale_settings.py

```diff
@@ -279,50 +486,104 @@ def autoscale_profile_list_timezones(cmd, client, offset=None, search_query=None
-            pass
-        return name.lower() != profile_name.lower()
-
-    autoscale_settings.profiles = [x for x in autoscale_settings.profiles if _should_retain_profile(x)]
+def autoscale_profile_show(cmd, autoscale_name, resource_group_name, profile_name):
+    autoscale_settings = AutoScaleShow(cli_ctx=cmd.cli_ctx)(command_args={
+        "resource_group": resource_group_name,
+        "autoscale_name": autoscale_name})
+    return _identify_profile_cg(autoscale_settings["profiles"], profile_name)
 
-    # if we removed the last "default" of a recurring pair, we need to preserve it
-    new_default, _ = build_autoscale_profile(autoscale_settings)
-    if not new_default:
-        autoscale_settings.profiles.append(default_profile)
 
-    autoscale_settings = client.create_or_update(resource_group_name, autoscale_name, autoscale_settings)
+class AutoScaleProfileDelete(_AutoScaleUpdate):
 
-
-def autoscale_rule_create(cmd, client, autoscale_name, resource_group_name, condition,
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.profile_name = AAZStrArg(
+            options=["--profile-name"],
+            help='Name of the autoscale profile.',
+            required=True,
+            registered=False,
+        )
+        return args_schema
+
+    def pre_instance_update(self, instance):
+        args = self.ctx.args
+        profile_name = args.profile_name.to_serialized_data()
+        instance = self.ctx.vars.instance
+        default_profile, _ = build_autoscale_profile_from_instance(instance)
+
+        def _should_retain_profile(profile):
+            name = profile.name.to_serialized_data()
+            try:
+                name = json.loads(profile.name.to_serialized_data())['for']
+            except ValueError:
+                pass
+            return name.lower() != profile_name.lower()
+
+        retained_profiles = [x for x in instance.properties.profiles if _should_retain_profile(x)]
+        instance.properties.profiles = retained_profiles
+
+        # if we removed the last "default" of a recurring pair, we need to preserve it
+        new_default, _ = build_autoscale_profile_from_instance(instance)
+        if not new_default:
+            instance.properties.profiles.append(default_profile)
+
+    def _output(self, *args, **kwargs):
+        from azure.cli.core.aaz import AAZUndefined
+        if has_value(self.ctx.vars.instance.properties.name):
+            self.ctx.vars.instance.properties.name = AAZUndefined
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+        return result
+
+
+def autoscale_profile_delete(cmd, autoscale_name, resource_group_name, profile_name):
+    AutoScaleProfileDelete(cli_ctx=cmd.cli_ctx)(command_args={
+        "autoscale_name": autoscale_name,
+        "resource_group": resource_group_name,
+        "profile_name": profile_name,
+    })
```

Comment:
You should return it. Specially when it's a LRO

Reply:
no need as previous func